compositor_pipeline/src/pipeline/input/rtp/depayloader.rs

```diff
@@ -101,24 +98,23 @@ impl VideoDepayloader {
                     return Ok(None);
                 }
 
-                if *last_timestamp == 0 {
-                    *last_timestamp = packet.header.timestamp;
-                }
-
-                if packet.header.timestamp == *last_timestamp {
+                if !packet.header.marker {
+                    // not at access unit border, buffer the packet
                     buffer.push(h264_chunk);
                     return Ok(None);
                 }
 
+                let mut data = buffer.concat();
+                data.extend(h264_chunk);
+
                 let new_chunk = EncodedChunk {
-                    data: buffer.concat().into(),
-                    pts: Duration::from_secs_f64(*last_timestamp as f64 / 90000.0),
+                    data: data.into(),
```

Comment:
```suggestion
                    data: buffer.concat().into(),
```
or
```suggestion
                    data: mem::take(buffer).concat().into(),
```
(and you don't need to assign to `*buffer` latter)